builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,104 @@
+    // Load ERC-20 interface of contract
+    IERC20 erc20 = IERC20(_erc20Contract);
+    // Get amount to transfer
+    uint256 amount = erc20.balanceOf(_arbitraryCallContract);
+    // Build the encoded call data
+    return abi.encodeWithSelector(IERC20.transfer.selector, _to, amount);
+}
+```
+
+Once you have the encoded call data, you could make an arbitrary call to the native ERC-20 precompile contract, set the value of the call to `0`, and pass in the call data in bytes:
+
+```
+function makeArbitraryCall(address _target, bytes calldata _bytes) public {
+    // Value: 0 does not protect against native ERC-20 precompile calls or XCM precompiles
+    (bool success,) = _target.call{value: 0}(_bytes);
+    require(success);
+}
+```
+
+The value of `0` will be overridden by the amount to be transferred as specified in the encoded call data, which in this example is the balance of the contract.
+
+### Whitelisting Safe Function Selectors {: #whitelisting-function-selectors }
+
+By whitelisting a specific function selector, you can control what functions can be executed and ensure only functions that are considered safe and do not call precompiles are allowed to be called. 
+
+To get the function selector to whitelist, you can [keccack256 hash](https://emn178.github.io/online-tools/keccak_256.html){target=_blank} the signature of the function.
+
+Once you have the whitelisted function selector, you can use inline assembly to get the function selector from the encoded call data and compare the two selectors using the [require function](https://docs.soliditylang.org/en/v0.8.17/control-structures.html#panic-via-assert-and-error-via-require){target=_blank}. If the function selector from the encoded call data matches the whitelisted function selector, you can make the call. Otherwise, an exception will be thrown.
+
+```
+function makeArbitraryCall(address _target, bytes calldata _bytes) public {
+    // Get the function selector from the encoded call data
+    bytes4 selector;
+    assembly {
+        selector := calldataload(_bytes.offset)
+    }
+
+    // Ensure the call data calls an approved and safe function
+    require(selector == INSERT-WHITELISTED-FUNCTION-SELECTOR);
+
+    // Arbitrary call
+    (bool success,) = _target.call(_bytes);
+    require(success);
+}
+```
+
+### Whitelisting Safe Contracts {: #whitelisting-safe-contracts}
+
+By whitelisting a specific target contract address in the function that can execute arbitrary call data, you can ensure that the call is considered safe, as the EVM will enforce that only whitelisted contracts can be called. This assumes that the contracts being called are not precompiles. If they are precompiles, you'll want to make sure that the contract making the call does not hold the native token or any XC-20. 
+
+Blacklisting contracts from arbitrary code execution is not considered safe, as other precompiles might be added in the future.
```

Comment:
We use a range for our precompiles starting with low values, so it could be feasible to blacklist the entire we might use (e.g. `0-4096` or so. We should discuss internally before documenting this as a good technique, though...